### DISCUSSION TITLE
How to present after dismiss with current playing song? Or only hide when I need?

### DISCUSSION THREAD

**Discussion by peetorres** [LeoNatan/LNPopupController#461] [General] [answered]
Hi Leo and community!

I'm starting a podcast project, and I'll use that library, but I faced some issues.

Situation: I have a tabBarController that contains three navigationControllers.

First o all, I play the song, starting the LNPopupFlow tabBarController.presentPopupBar(withContentViewController), right? that's a PlayerViewController. And I'll present on tabBar.

When user push a new controller from navigationController, such as Settings, naturally, popupController and tabBarController appears... because NavigationController is embed of tabBar, right? 

When I push a newController, if I handle hidesBottomBarWhenPushed = true, I can hide the tabBar, but popup appears...
If I dismissPopupBar, i'll dismiss entire behaviors of the component, right? How can I only hide that on this situation?

I tried to only change visibility, but such as you specified, it's not a correct way to hide.

I'm using tabBarController.presentPopupBar() , then, when I want to "hide", I wrote that function:

```
internal func songHandler(_ song: SongInformation) {
        currentPlaying = PlayerViewController(with: song)
        tabBarController?.presentPopupBar(withContentViewController: currentBookPlaying!,
                                          animated: false,
                                          completion: nil)
    }
```

```
internal func handleBottomElementsVisibility(with viewController: UIViewController) {
        viewController.hidesBottomBarWhenPushed = true
        tabBarController?.dismissPopupBar(animated: true, completion: nil)
}
```

So, when my controllers didAppear, I wrote that func:

```
internal func shouldHandlePopupBarVisibility() {
        DispatchQueue.main.async { [weak self] in
            guard let currentPlaying = self?.currentBookPlaying else { return }
            self?.tabBarController?.presentPopupBar(withContentViewController: currentPlaying,
                                                    animated: false,
                                                    completion: nil)
        }
}
```

I imagine that there's a simplest way to only hide that component when you need that kind of behavior... but I didn't found it.

Imagine Spotify, you could push a new screen (without component), continue playing and some screens not show the component... open is the entire player, close too, right?

What is the best approach for that? Do you wrote a function to handle position or something like that to only hide of screen? I tried change frame height, tried lots of things :(.

**Comment by LeoNatan** [comment, marked as answer]
Hello,

There is no perfect answer here.
- If one of your tabs is the settings tab, you can present and dismiss the popup bar depending on the selected tab, but then you need to be careful if you have logic elsewhere that presents the bar again. Maybe you can subclass the tab bar controller, and override the popup bar presentation methods to do the bookkeeping there.
- If you have a button that pushes a settings controller, consider presenting it as modal, rather than pushing. That is the more correct thing to do from a HIG standpoint, and this is what Apple does in their Music and Podcasts apps. If you prefer to push, I don't think there is something you can do to have the perfect experience. The framework is intended to not hide the popup bar when `hidesBottomBarWhenPushed` is enabled, but rather move the bar to the bottom. I think you will have to fight the system a lot to hide the bar on push and make it look well (no odd animation jumps, for example).

I suggest presenting you settings modally.

**Reply by peetorres** [reply]
Thank you to the answer Leo!

Designer changed the behavior enabling display of popupbar always when playing.

**Reply by LeoNatan** [reply]
Great!
When you get your app out, if want, you can link to it here: https://github.com/LeoNatan/LNPopupController/discussions/454 😀
